test/testdata/infer/generics/bounds.rb

```diff
@@ -12,56 +12,53 @@ class A1
 
   # should fail: T2 mentions another type member
   T2 = type_member(lower: T1)
-     # ^^^^^^^^^^^^^^^^^^^^^^ error: Only `:fixed` type members are supported
 
   # should fail: T3 mentions another type member
   T3 = type_member(lower: T2, upper: T1)
-     # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Only `:fixed` type members are supported
+     # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: `A1::T2` is not a subtype of `A1::T1`
 
   # should fail: T4 mentions another type member (this message could be better)
   T4 = type_member(fixed: T1)
-# ^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Expression does not have a fully-defined type
 
   # should fail: the bounds are invalid
   T5 = type_member(lower: Animal, upper: Persian)
-     # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Only `:fixed` type members are supported
+     # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: `Animal` is not a subtype of `Persian`
 
-  # should fail: type member used as an argument to another type (this message
-  # could be better)
+  # should fail: type member used as an argument to another type
   T6 = type_member(fixed: T::Array[T1])
-# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Expression does not have a fully-defined type
 
   # should fail: type member used as an argument to another type
   T7 = type_member(lower: T::Array[T1])
-     # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Only `:fixed` type members are supported
 
   # should fail: both bounds and a fixed type are specified
   T8 = type_member(fixed: Cat, lower: Persian, upper: Animal)
-     # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Type member is defined with bounds and `:fixed`
-     # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Only `:fixed` type members are supported
+     # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Type member is defined with bounds and `fixed`
 
   # should pass: just an alias to a type member
   T9 = T1
 
   # should fail: still using a type member in the definition of another
   T10 = type_member(fixed: T9)
-# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Expression does not have
 
   # should fail: using a type alias in bounds
   T11 = type_member(upper: T9)
-      # ^^^^^^^^^^^^^^^^^^^^^^ error: Only `:fixed` type members are supported
 
   # should fail: multiple upper and lower bounds specified
   T12 = type_member(lower: Integer, lower: String, upper: BasicObject, upper: TrueClass)
```

Comment:
I'm unsure if we should check this, we don't do this kind of checking for anything else.
Also, Ruby warns on this:
```
[2] pry(main)> foo(a: 1, a: 2)
(pry):5: warning: key :a is duplicated and overwritten on line 5
```

Reply:
I've removed this additional checking.